Description: remove invalid free() call
 The data pointer is not allocated separatedly, it's the struct that needs to
 be free()ed.
 Originally reported by lixiaokeng via
 https://www.spinics.net/lists/linux-nfs/msg87598.html with a slightly different
 fix proposal.
Author: Andreas Hasenack <andreas@canonical.com>
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1022185
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/nfs-utils/+bug/1979885
Forwarded: https://lore.kernel.org/linux-nfs/CANYNYEG=utJ2pe+FtMWh8O+dz63R2wbzOC7ZVrvoqD=U04WL5g@mail.gmail.com/T/#u
Last-Update: 2022-10-20
diff --git a/utils/blkmapd/device-discovery.c b/utils/blkmapd/device-discovery.c
index 2736ac89..b3786a1f 100644
--- a/utils/blkmapd/device-discovery.c
+++ b/utils/blkmapd/device-discovery.c
@@ -188,7 +188,6 @@ static void bl_add_disk(char *filepath)
 
 	if (disk && diskpath) {
 		if (serial) {
-			free(serial->data);
 			free(serial);
 		}
 		return;
@@ -229,7 +228,6 @@ static void bl_add_disk(char *filepath)
 			disk->valid_path = path;
 		}
 		if (serial) {
-			free(serial->data);
 			free(serial);
 		}
 	}
@@ -242,7 +240,6 @@ static void bl_add_disk(char *filepath)
 		free(path);
 	}
 	if (serial) {
-		free(serial->data);
 		free(serial);
 	}
 	return;
